src/librustdoc/html/static/main.js

```diff
@@ -935,37 +935,39 @@
                 var previousTitle = document.title;
 
                 window.onpopstate = function(e) {
-                    var params = getQueryStringParams();
-                    // When browsing back from search results the main page
-                    // visibility must be reset.
-                    if (!params.search) {
-                        var main = document.getElementById('main');
-                        if (hasClass(main, 'content')) {
-                            removeClass(main, 'hidden');
-                        }
-                        var search = document.getElementById('search');
-                        if (hasClass(main, 'content')) {
-                            addClass(main, 'hidden');
+                    if (location.href.indexOf(".html?search=") !== -1) {
+                        var params = getQueryStringParams();
+                        // When browsing back from search results the main page
+                        // visibility must be reset.
+                        if (!params.search) {
+                            var main = document.getElementById('main');
+                            if (hasClass(main, 'content')) {
+                                removeClass(main, 'hidden');
+                            }
+                            var search = document.getElementById('search');
+                            if (hasClass(main, 'content')) {
+                                addClass(main, 'hidden');
+                            }
                         }
+                        // Revert to the previous title manually since the History
+                        // API ignores the title parameter.
+                        document.title = previousTitle;
+                        // When browsing forward to search results the previous
+                        // search will be repeated, so the currentResults are
+                        // cleared to ensure the search is successful.
+                        currentResults = null;
+                        // Synchronize search bar with query string state and
+                        // perform the search. This will empty the bar if there's
+                        // nothing there, which lets you really go back to a
+                        // previous state with nothing in the bar.
+                        document.getElementsByClassName('search-input')[0].value = params.search;
```

Comment:
If `params.search` is `undefined` this should set the value to `""` not `"undefined"` like it currently does.